socs/agents/smurf_timing_card/agent.py

```diff
@@ -0,0 +1,93 @@
+import argparse
+import os
+import time
+
+import txaio
+from epics import PV
+from ocs import ocs_agent, site_config
+from ocs.ocs_twisted import Pacemaker
+
+diagnostic_root = 'TPG:SMRF:1'
+regs = [
+    'COUNTPLL', 'COUNT186M', 'COUNTSYNCERR', 'COUNTINTV', 'COUNTBRT',
+    'COUNTTXCLK', 'DELTATXCLK', 'RATETXCLK'
+]
+regs_full = {
+    r: f'{diagnostic_root}:{r}' for r in regs
+}
+
+
+class SmurfTimingCardAgent:
+    """
+    Agent to monitor diagnostic variables for the SMuRF Timing Card.
+    """
+
+    def __init__(self, agent, args):
+        self.agent = agent
+        self.log = agent.log
+        self.sleep_time = args.sleep_time
+        self.pvs = {
+            k: PV(r) for k, r in regs_full.items()
+        }
+        self.timeout = args.timeout
+        self.use_monitor = args.use_monitor
+        agent.register_feed('diagnostics', record=True)
+
+    def run(self, session, params):
```

Comment:
Sorry, one more request -- this could use an example in the docstring of what's stored in `session.data`. Can be a copy + paste of the structure from the running agent if you have it online.